File: test/test_assign/files/exception-correct/student/tests/q1.py
from otter.test_files import test_case

OK_FORMAT = False

name = "q1"
points = None
all_or_nothing = True

@test_case(points=2, hidden=False, 
    success_message="Congrats you passed this test case!", 
    failure_message="This is not an int.")
def test_type(x):
    assert isinstance(x, int)
    assert 0 < x < 100

File: test/test_run/files/qmd-autograder/source/tests/q1.R
test = list(
  name = "q1",
  cases = list(
    ottr::TestCase$new(
      hidden = FALSE,
      name = NA,
      points = 1,
      code = {
        testthat::expect_true(is.numeric(x))
      }
    ),
    ottr::TestCase$new(
      hidden = FALSE,
      name = NA,
      points = 1,
      code = {
        testthat::expect_true(0 < x)
        testthat::expect_true(x < 100)
      }
    ),
    ottr::TestCase$new(
      hidden = TRUE,
      name = NA,
      points = 1,
      code = {
        testthat::expect_equal(x, 2)
      }
    ),
    ottr::TestCase$new(
      hidden = TRUE,
      name = "q1d",
      points = 2,
      success_message = "congrats",
      code = {
        testthat::expect_equal(as.character(x), "2")
      }
    )
  )
)